build/org.eclipse.pde.build/src/org/eclipse/pde/internal/build/site/PDEState.java

```diff
@@ -462,6 +461,50 @@ public void resolveState() {
 		}
 	}
 
+	@SuppressWarnings("restriction")
+	public static String querySystemPackages(IExecutionEnvironment environment) {
+		// Copy of org.eclipse.pde.internal.core.TargetPlatformHelper.querySystemPackages()
+		IVMInstall vm = bestVmInstallFor(environment);
+		if (vm == null || !JavaRuntime.isModularJava(vm)) {
+			return null;
+		}
+		String release = environment.getProfileProperties().getProperty(JavaCore.COMPILER_COMPLIANCE);
+		try {
+			Collection<String> packages = new TreeSet<>();
+			String jrtPath = "lib/" + org.eclipse.jdt.internal.compiler.util.JRTUtil.JRT_FS_JAR; //$NON-NLS-1$
+			String path = new File(vm.getInstallLocation(), jrtPath).toString(); // $NON-NLS-1$
+			var jrt = org.eclipse.jdt.internal.core.builder.ClasspathLocation.forJrtSystem(path, null, null, release);
+			for (String moduleName : jrt.getModuleNames(null)) {
+				for (var packageExport : jrt.getModule(moduleName).exports()) {
```

Comment:
Please don't dereference `jrt.getModule(moduleName)`  unconditionally. This doesn't guarantee to return non-null value.

See https://github.com/eclipse-pde/eclipse.pde/pull/286

Reply:
Thanks for the hint. Applied the same patch.